velox/expression/CastExpr.h

```diff
@@ -259,6 +259,8 @@ class CastExpr : public SpecialForm {
       VectorPtr& result,
       const date::time_zone* timeZone = nullptr);
 
+  virtual int32_t getCastFromDateStringResult(const StringView& inputString);
```

Comment:
I wonder if it would be cleaner to define a separate class CastHooks that allows to customize Cast behavior and pass an instance of that class to CastExpr's constructor.

Reply:
@rui-mo Based on your analysis of cast differences, what other hooks would we need to add here? Would it be possible to prototype full functionality to see how it turns out?

Reply:
Yes, it makes sense. I will try to create CastHooks, and enhance the functionality of SparkCastExpr.

Reply:
Hi @mbasmanova, I created CastHooks, and implemented most of the functionality described in https://github.com/facebookincubator/velox/issues/4876#issuecomment-1838011064. Could you help check the added hooks, and determine if we should continue with current idea?

Two missing points are 1) support more Spark-allowed patterns for cast from string to timestamp (better to go into a separate PR). 2) handle white-spaces differently for cast from string to decimal (should be handled in https://github.com/facebookincubator/velox/pull/5307).

Reply:
@rui-mo Sounds great. Let me take a look. In the meantime, would you update PR description? It seems outdated.